echopop/nwfsc_feat/get_proportions.py

```diff
@@ -1,162 +1,1529 @@
+    pd.Series
+        NASC proportions by strata, representing the fraction of total acoustic
+        backscatter contributed by the filtered group
+
+    Notes
+    -----
+    The target strength calculation follows: TS = slope * log10(length) + intercept
+    Backscatter cross-section (sigma_bs) = 10^(TS/10)
+
+    Examples
+    --------
+    >>> # Get NASC proportions for age-1 fish
+    >>> nasc_props = get_nasc_proportions_slice(
+    ...     number_proportions=aged_data,
+    ...     include_filter={"age_bin": [1]}
+    ... )
+    """
+
+    # Get length values from length bins
+    length_vals = number_proportions["length_bin"].apply(lambda x: x.mid).astype(float).unique()
+
+    # Compute equivalent sigma_bs using target strength-length regression
+    sigma_bs_equiv = 10 ** (
+        (
+            ts_length_regression_parameters["slope"] * np.log10(length_vals)
+            + ts_length_regression_parameters["intercept"]
+        )
+        / 10.0
+    )
+
+    # Create pivot table for total population (all groups)
+    aggregate_table = utils.create_pivot_table(
+        number_proportions,
+        index_cols=["length_bin"],
+        strat_cols=stratify_by,
+        value_col="proportion",
+    )
+
+    # Calculate total weighted sigma_bs for all strata
+    aggregate_weighted_sigma_bs = (aggregate_table.T * sigma_bs_equiv).T.sum(axis=0)
+
+    # Create pivot table including filter dimensions
+    filtered_population_table = utils.create_pivot_table(
+        number_proportions,
+        index_cols=list(include_filter.keys()) + ["length_bin"],
+        strat_cols=stratify_by,
+        value_col="proportion",
+    )
 
-        # TODO: sketch of ds_proportions
-        - dimensions: stratum, sex, length_bin, age_bin
-        - variables:
-          - abundance_unaged: (stratum, sex, length_bin)
-          - abundance_aaged: (stratum, sex, length_bin, age_bin)
-          - biomass_aged: (stratum, sex, length_bin, age_bin)
-          - biomass_unaged: (stratum, sex, length_bin)
+    # Apply filter to extract target group
+    target_group_table = utils.apply_filters(
+        filtered_population_table, include_filter=include_filter
+    )
 
+    # Aggregate target group over length and strata dimensions
+    target_group_aggregated = target_group_table.unstack("length_bin").sum().unstack(stratify_by)
 
+    # Calculate weighted sigma_bs for target group
+    target_weighted_sigma_bs = (target_group_aggregated.T * sigma_bs_equiv).T.sum(axis=0)
+
+    # Calculate and return proportions
+    return target_weighted_sigma_bs / aggregate_weighted_sigma_bs
+
+
+def get_number_proportions_slice(
+    number_proportions: pd.DataFrame,
+    stratify_by: List[str],
+    exclude_filter: Dict[str, Any] = {},
+    include_filter: Dict[str, Any] = {},
+) -> Union[pd.Series, pd.DataFrame]:
     """
-    ds_proportions: xr.Dataset
-    return ds_proportions
+    Extract number proportions for a specific population slice with flexible filtering.
```

Comment:
Making the description consistent with the others.
```suggestion
    Extract number proportions for a specific slice of the population.
```